resources/iam/users.go

```diff
@@ -237,46 +245,57 @@ func (c *Client) transformReportUser(ctx context.Context, reportUser *ReportUser
 
 	}
 
-	switch strings.ToLower(reportUser.PasswordEnabled) {
-	case "false":
-		passwordEnabled := false
-		res.PasswordEnabled = &passwordEnabled
-	case "true":
-		passwordEnabled := true
-		res.PasswordEnabled = &passwordEnabled
-	}
+	if reportUser != nil {
+		switch strings.ToLower(reportUser.PasswordEnabled) {
+		case "false":
+			passwordEnabled := false
+			res.PasswordEnabled = &passwordEnabled
+		case "true":
+			passwordEnabled := true
+			res.PasswordEnabled = &passwordEnabled
+		}
 
-	passwordLastUsed, err := time.ParseInLocation(time.RFC3339, reportUser.PasswordLastUsed, location)
-	if err == nil {
-		res.PasswordLastUsed = &passwordLastUsed
-	}
-	passwordLastChanged, err := time.ParseInLocation(time.RFC3339, reportUser.PasswordLastChanged, location)
-	if err == nil {
-		res.PasswordLastChanged = &passwordLastChanged
-	}
+		passwordLastUsed, err := time.ParseInLocation(time.RFC3339, reportUser.PasswordLastUsed, location)
+		if err == nil {
+			res.PasswordLastUsed = &passwordLastUsed
+		}
+		passwordLastChanged, err := time.ParseInLocation(time.RFC3339, reportUser.PasswordLastChanged, location)
+		if err == nil {
+			res.PasswordLastChanged = &passwordLastChanged
+		}
 
-	passwordNextRotation, err := time.ParseInLocation(time.RFC3339, reportUser.PasswordNextRotation, location)
-	if err == nil {
-		res.PasswordNextRotation = &passwordNextRotation
-	}
+		passwordNextRotation, err := time.ParseInLocation(time.RFC3339, reportUser.PasswordNextRotation, location)
+		if err == nil {
+			res.PasswordNextRotation = &passwordNextRotation
+		}
 
-	lastRotated1, err := time.ParseInLocation(time.RFC3339, reportUser.AccessKey1LastRotated, location)
-	if err == nil && len(res.AccessKeys) > 0 {
-		res.AccessKeys[0].LastRotated = &lastRotated1
-	}
+		lastRotated1, err := time.ParseInLocation(time.RFC3339, reportUser.AccessKey1LastRotated, location)
+		if err == nil && len(res.AccessKeys) > 0 {
+			res.AccessKeys[0].LastRotated = &lastRotated1
+		}
 
-	lastRotated2, err := time.ParseInLocation(time.RFC3339, reportUser.AccessKey2LastRotated, location)
-	if err == nil && len(res.AccessKeys) > 1 {
-		res.AccessKeys[1].LastRotated = &lastRotated2
+		lastRotated2, err := time.ParseInLocation(time.RFC3339, reportUser.AccessKey2LastRotated, location)
+		if err == nil && len(res.AccessKeys) > 1 {
+			res.AccessKeys[1].LastRotated = &lastRotated2
+		}
 	}
 
 	return &res, nil
 }
 
-func (c *Client) transformReportUsers(ctx context.Context, values []*ReportUser) ([]*User, error) {
+func getMatchingReportUser(reportUsers []*ReportUser, user types.User) *ReportUser {
+	for _, reportUser := range reportUsers {
+		if user.Arn == &reportUser.ARN {
```

Comment:
Strange that it worked because looks like you are comparing pointers. It should be the other way around. `*user.Arn == reportUser.ARN`